### ISSUE TITLE
FR: Correctly reflect ACL restrictions on device lists in clients

### ISSUE THREAD

**Issue by agherzan** [tailscale/tailscale#9065]
### What are you trying to do?

If I set an ACL that restricts connections between device1 (user1) to device2 (user2), user2 will also get the device1 listed in the client's interface (for example, in the mobile app) even though a connection is not allowed.

This makes the following ACL rule impossible to do without exposing node information:
//allow connections from all devices owned user1 to all the devices owned by user2"
{
"action": "accept",
"src": ["user1@example.com)"],
"dst": ["user2@example.com:*"],
}

Even though the rule only allowed connection from the devices owned by user1 to those owned by user2, user2 seels all the devices owned by user1 in the tailscale client app.

### How should we solve this?

The ACL restriction should reflect similarly to the visibility of nodes in the apps.

### What is the impact of not solving this?

Users get advertised devices that they can't access, and also it shares device node information even when not allowing access to them.

**Comment by DentonGentry** [comment]
This is intentional: if neither node can connect to the other then we remove it from the netmap altogether. If either node can connect to the other, in either direction, the connection information is populated in the netmap and the existence of those clients is visible to the other. By design, there are not invisible nodes which can connect to your node.

**Comment by agherzan** [comment]
Makes sense. I know this is due to keys exchange, but the UX/UI should clarify this. Maybe we should have a special section for these devices (that can be expanded or so). I haven't thought too much about it from the UX perspective, but this leads to confusion when you are sharing a tailnet with other users.

**Comment by 0xSterny** [comment]
We are issuing out single use keys to client VMs to join our Tailscale network under specific client tags. If a client enters "tailscale status", they are able to see all other clients within our Tailnet, regardless of ability to access via ACLs. Is there any way to prevent PII leakage when entering "tailscale status"?
